internal/api/e2e/framework/server.go

```diff
@@ -139,24 +188,35 @@ func CreateServer(t *testing.T, conf apiConfig.Config) (*httptest.Server, *httpe
 	return server, httpexpect.WithConfig(httpexpect.Config{
 		BaseURL:  server.URL,
 		Reporter: httpexpect.NewAssertReporter(t),
-	}), persistenceManager
+	}), dependencyManager
 }
 
 func WithServer(t *testing.T, testFunc func(*httptest.Server, *httpexpect.Expect, dependency.PersistenceManager) []modelAPI.Entity) {
 	conf := DefaultConfig()
-	server, expect, persistenceManager := CreateServer(t, conf)
-	defer persistenceManager.GetPersesDAO().Close()
+	server, expect, dependencyManager := CreateServer(t, conf)
+	defer dependencyManager.Persistence().GetPersesDAO().Close()
+	defer server.Close()
+	entities := testFunc(server, expect, dependencyManager.Persistence())
+	ClearAllKeys(t, dependencyManager.Persistence().GetPersesDAO(), entities...)
+}
+
+func WithServerAuthConfig(t *testing.T, testFunc func(*httptest.Server, *httpexpect.Expect, dependency.Manager, string) []modelAPI.Entity) {
```

Comment:
I'm not fan of the way the test code is closed. 
I'd prefer something a bit more modular. Something in that spirit

```go
e2eframework.WithServerConfig(t, e2eframework.DefaultAuthConfig(), ...
    token, usrEntity, globalRole, globalRoleBinding := e2eframework.BuildUserWithGlobalRoleAndLogin(expect, "alice", "admin", dependencyManager)
}
```
There is still the only native login supported that I'm not fan of. 

I think I can try to make a proposal of update it will be better than speaking.